src/rgw/rgw_rados.h

```diff
@@ -1313,10 +1313,12 @@ class RGWRados
   RGWDataChangesLog *data_log;
 
   virtual ~RGWRados() {
-    for (uint32_t i=0; i < num_rados_handles; i++) {
-      if (rados[i]) {
-        rados[i]->shutdown();
-        delete rados[i];
+    if (rados){
```

Comment:
I don't think so.
1. if rados has been set to NULL, rados[i] also is an invalid reference.